python/felis/metadata.py

```diff
@@ -0,0 +1,500 @@
+        apply_schema_to_tables : `bool`, optional
+            Whether to apply the schema name to the tables.
+        """
+        self.schema = schema
+        if not apply_schema_to_metadata:
+            logger.debug("Schema name will not be applied to metadata")
+        if not apply_schema_to_tables:
+            logger.debug("Schema name will not be applied to tables")
+        self.metadata = MetaData(schema=schema.name if apply_schema_to_metadata else None)
+        self._objects: dict[str, Any] = {}
+        self.apply_schema_to_tables = apply_schema_to_tables
+
+    def reset(self, schema: Schema) -> None:
+        """Reset the builder with a new schema.
+
+        Parameters
+        ----------
+        schema : `felis.datamodel.Schema`
+            The new schema object to build the SQA metadata from.
+        """
+        self.schema = schema
+        self.metadata = MetaData(schema=self.schema.name)
+        self._objects = {}
+
+    def build(self) -> MetaData:
+        """Build the SQA tables and constraints from the schema."""
+        self.build_tables()
+        self.build_constraints()
+        return self.metadata
+
+    def build_tables(self) -> None:
+        """Build the SQA tables from the schema.
+
+        Notes
+        -----
+        This is the main function for building the SQA tables from the schema
+        including objects within tables such as constraints, primary keys,
+        and indices, which each have their own dedicated sub-functions.
+        """
+        for table in self.schema.tables:
+            self.build_table(table)
+            if table.primaryKey:
+                primary_key = self.build_primary_key(table.primaryKey)
+                self._objects[table.id].append_constraint(primary_key)
+
+    def build_primary_key(self, primary_key_columns: str | list[str]) -> PrimaryKeyConstraint:
+        """Build a SQA `PrimaryKeyConstraint` from a single column ID or a list
+        or them.
+
+        Parameters
+        ----------
+        primary_key_columns : `str` or `list` of `str`
+            The column ID or list of column IDs from which to build the primary
+            key.
+
+        Returns
+        -------
+        primary_key: `PrimaryKeyConstraint`
+            The SQA primary key constraint object.
+        """
+        columns: list[Column] = []
+        if isinstance(primary_key_columns, str):
+            columns.append(self._objects[primary_key_columns])
+        else:
+            columns.extend([self._objects[column_id] for column_id in primary_key_columns])
+        return PrimaryKeyConstraint(*columns)
+
+    def build_table(self, table_obj: dm.Table) -> None:
+        """Build a SQA `Table` from a `felis.datamodel.Table` and add it to the
+        `MetaData` object.
+
+        Parameters
+        ----------
+        table_obj : `felis.datamodel.Table`
+            The table object to build the SQA table from.
+        """
+        # Process mysql table options.
```

Comment:
The docstring says nothing about mysql but this method seems to be using mysql-specific names. Maybe enhance the code comment here saying why postgres doesn't need this tweak.

Reply:
> The docstring says nothing about mysql but this method seems to be using mysql-specific names. Maybe enhance the code comment here saying why postgres doesn't need this tweak.

I will add documentation on this. Postgres does not have similar table options that require special handling.